src/core/public/chrome/ui/header/collapsible_nav.tsx

```diff
@@ -71,6 +71,29 @@ function getOrderedCategories(
   );
 }
 
+function getMergedNavLinks(
+  orderedCategories: string[],
+  uncategorizedLinks: ChromeNavLink[],
+  categoryDictionary: ReturnType<typeof getAllCategories>
+): Array<string | ChromeNavLink> {
+  const uncategorizedLinksWithOrder = sortBy(
+    uncategorizedLinks.filter((link) => link.order !== null),
+    'order'
+  );
+  const uncategorizedLinksWithoutOrder = uncategorizedLinks.filter((link) => link.order === null);
+  const orderedCategoryWithOrder = orderedCategories
+    .filter((categoryName) => categoryDictionary[categoryName]?.order !== null)
+    .map((categoryName) => ({ categoryName, order: categoryDictionary[categoryName]?.order }));
+  const orderedCategoryWithoutOrder = orderedCategories.filter(
+    (categoryName) => categoryDictionary[categoryName]?.order === null
+  );
+  const mergedNavLinks = sortBy(
+    [...uncategorizedLinksWithOrder, ...orderedCategoryWithOrder],
+    'order'
+  ).map((navLink) => ('categoryName' in navLink ? navLink.categoryName : navLink));
+  return [...mergedNavLinks, ...orderedCategoryWithoutOrder, ...uncategorizedLinksWithoutOrder];
```

Comment:
All LGTM, but here we'd better add a comment to say that we sort links with category before those without category by `convention` that categoried links appear before uncategoried ones.